CodeReviewStandards/rfc-008-review-checklist.md

```diff
@@ -0,0 +1,45 @@
+# Pull Request Standards
+
+## Summary
+
+A checklist designed to help facilitate code reviews by outlining a pre-review checklist to follow for the raiser of the pull request, and a reviewer checklist.
+
+## Problem
+
+Pull requests follow no set format and vary between projects, which in turn causes reviews to take longer and require more context switching. Code review and commentary often happens 'offline', which results in reduced context and fewer points of reference when reviewing changes later.
+
+## Proposal
+
+Before raising a pull request, the pull request SHOULD meet the following criteria:
+
+ - [ ] Pull request description outlines _what has changed_
+ - [ ] Pull request description outlines _why the change was made_
+ - [ ] If the project has a linter, it SHOULD not report any issues with the pull request
+ - [ ] Code follows language specific conventions
+ - [ ] Code follows project specific conventions
```

Comment:
What if language and project conventions conflict? (I'm assuming project > language but might be worth making clear)

Reply:
Thinking of a Python project which might not choose to follow PEP8

Reply:
Every checkpoint is SHOULD, which means it can be discounted if there is justification (python project omitting PEP8 in favour of a project linter is a perfect example). I'd say that project will always take precedence over language convention and I'll make it clearer (unless anyone strongly feels language convention > project)